### PR TITLE
Navigate to #admin on workspace creation

### PR DESCRIPTION
**Pull request by esh-g**
### Fixed Issues

$ https://github.com/Expensify/App/issues/21408
PROPOSAL: https://github.com/Expensify/App/issues/21408#issuecomment-1604334247

### Tests

**Test 1**
1. Login from an account with no workspace (or delete all workspaces)
2. Click on the `+` icon and create new workspace.
3. On fullscreen, verify that we navigate to #admins page report and also to the workspace details on the right.
4. On Small screen, verify that we are not shown the admin chat but are navigated to the details of the workspace.

**Test 2**
1. Go to settings -> workspace.
2. Click on "New Workspace"
3. On fullscreen, verify that we navigate to #admins page report and also to the workspace details on the right.
4. On Small screen, verify that we are not shown the admin chat but are navigated to the details of the workspace.

**Test 3** (For web and mWeb)
1. Login to Old dot.
2. Go to settings -> policies -> groups.
3. Click on new policy.
4. Copy the URL and close the tab.
5. Open a new tab and paste the URL replacing `new.expensify.com` with `localhost:8080`.
6. On fullscreen, verify that we navigate to #admins page report and also to the workspace details on the right.
7. On Small screen, verify that we are not shown the admin chat but are navigated to the details of the workspace.

- [x] Verify that no errors appear in the JS console

### Offline tests

**Test 2** (Only this test required for offline)
1. Go to settings -> workspace.
2. Click on "New Workspace"
3. On fullscreen, verify that we navigate to #admins page report and also to the workspace details on the right.
4. On Small screen, verify that we are not shown the admin chat but are navigated to the details of the workspace.

### QA Steps

**Test 1**
1. Login from an account with no workspace (or delete all workspaces)
2. Click on the `+` icon and create new workspace.
3. On fullscreen, verify that we navigate to #admins page report and also to the workspace details on the right.
4. On Small screen, verify that we are not shown the admin chat but are navigated to the details of the workspace.

**Test 2**
1. Go to settings -> workspace.
2. Click on "New Workspace"
3. On fullscreen, verify that we navigate to #admins page report and also to the workspace details on the right.
4. On Small screen, verify that we are not shown the admin chat but are navigated to the details of the workspace.

**Test 3** (For web and mWeb)
1. Login to Old dot.
2. Go to settings -> policies -> groups.
3. Click on new policy.
4. Copy the URL and close the tab.
5. Open a new tab and paste the URL replacing `new.expensify.com` with `staging.new.expensify.com`
8. On fullscreen, verify that we navigate to #admins page report and also to the workspace details on the right.
9. On Small screen, verify that we are not shown the admin chat but are navigated to the details of the workspace.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/77237602/38bd8278-2359-488e-8391-0868e12479fa

https://github.com/Expensify/App/assets/77237602/2c578d60-fa7b-4a49-bad6-bcb968b36045

https://github.com/Expensify/App/assets/77237602/41f2e014-537b-4dfd-a47f-4d9ba3e54b44

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/77237602/65abb8fc-0944-426a-9ede-7602b277cf26

https://github.com/Expensify/App/assets/77237602/b5cf4d27-7002-43e3-b2d7-06be68254d70

https://github.com/Expensify/App/assets/77237602/8067e320-af25-4bb8-94ae-4e57bf74b94a

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/77237602/ba2bae70-209e-4b6c-b20c-8ec37a755d41

https://github.com/Expensify/App/assets/77237602/6fec3c01-9bf4-40cb-8ad1-495b610f6ecc

https://github.com/Expensify/App/assets/77237602/a7c250c5-1628-4fb0-af5e-5c2dde2ddcc7

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/77237602/630172c1-6a92-4432-b32a-e55787e7fc35

https://github.com/Expensify/App/assets/77237602/0d86d0c2-b47b-4c5f-a4ee-901b91e9465d

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/77237602/02773da1-ac04-479b-a8b0-8734eafd4126

https://github.com/Expensify/App/assets/77237602/f4533c29-faf7-4054-8cfb-f85098d86294

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/77237602/cbcc3922-99db-4031-9738-acb20c6ca63b

https://github.com/Expensify/App/assets/77237602/062d582d-a00f-4e56-b1eb-eb09d58a2e2a

</details>

### CODE REVIEW

**Comment by esh-g** [comment]
bump @abdulrahuman5196 I hope this looks good!

**Comment by abdulrahuman5196** [comment]
Sorry for the delay. Will complete the review in the morning.

**Comment by abdulrahuman5196** [comment]
@esh-g There is an another createWorkspace ingress here which is missed in the PR - https://github.com/Expensify/App/blob/9e5d4ee9ad4b0711ddb119009d49fa872e662f14/src/libs/actions/App.js#L225

**Comment by esh-g** [comment]
@abdulrahuman5196 Then would it be fine to pass the `shouldNavigateToAdminChat` argument rom `AuthScreen.js (App.setUpPoliciesAndNavigate)` --> `actions/App.js` (Policy.createWorkspace). In there:
https://github.com/Expensify/App/blob/9b205fbb26aeafea690b62412d00fd1aa5529bde/src/libs/Navigation/AppNavigator/AuthScreens.js#L138
```js
 App.setUpPoliciesAndNavigate(this.props.session, !this.props.isSmallScreenWidth);
 ```

**Comment by esh-g** [comment]
@abdulrahuman5196 This is the only case where the function `setUpPoliciesAndNavigate` is being called as well, so it can be easily changed without affecting other places.

**Comment by abdulrahuman5196** [comment]
Yes. Its fine.

**Comment by esh-g** [comment]
@abdulrahuman5196 Sorry, I thought we only needed to add tests for the main bug.
I edited the Tests and am now creating the videos for the tests... But for Test 3, I'm not sure that is testable on native platforms...(because it involves manipulating URLs) Is it alright if I leave those out?

**Comment by esh-g** [comment]
@abdulrahuman5196 I have now added videos for all platforms as well (except the ones mentioned in the above comment)

**Comment by abdulrahuman5196** [comment]
@esh-g For the old dot migration flow,

For some reason the LHN becomes unresponsive for me, but the same doesn't happen on main.


https://github.com/Expensify/App/assets/46707890/17307193-c8e1-46ec-8ad6-db68ce73b368

**Comment by esh-g** [comment]
@abdulrahuman5196 I merged main into this branch and it goes away.

**Before**

https://github.com/Expensify/App/assets/77237602/7195ff16-b23a-4eba-af24-f746d2bdf58c

**After**

https://github.com/Expensify/App/assets/77237602/cdfb1aff-8a94-4ccd-a217-19f2a6b21397

**Comment by esh-g** [comment]
@abdulrahuman5196 Do let me know if that fixes it for you 😇

**Comment by esh-g** [comment]
Bump @abdulrahuman5196

**Comment by esh-g** [comment]
@abdulrahuman5196 Seems like another PR removed the logic to navigate to the admins chat as well as policy details from the `Policy.createWorkspace` and deferred the navigation to `App.createWorkspaceAndNavigateToIt`. Should I just replace the `Navigation.setParams` there like I did in this solution?

**Comment by abdulrahuman5196** [comment]
@esh-g Yes. Its only the position of the code that has changed. The same solution should apply AFAIK

**Comment by esh-g** [comment]
@abdulrahuman5196 I have made the required changes!

**Comment by esh-g** [comment]
bump @abdulrahuman5196

**Comment by abdulrahuman5196** [comment]
Hi, will work on the review today

**Comment by abdulrahuman5196** [comment]
Working on review now

**Comment by abdulrahuman5196** [comment]
@esh-g Test 2 alone will be applicable for offline test as well

**Comment by abdulrahuman5196** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>






https://github.com/Expensify/App/assets/46707890/482afbbd-83c0-4653-8a48-572a7b74a4bf


https://github.com/Expensify/App/assets/46707890/3a85a5d6-b100-4227-b448-59c3ac40c665

</details>

<details>
<summary>Mobile Web - Chrome</summary>




https://github.com/Expensify/App/assets/46707890/e421fbe0-955c-4691-b433-2f882aee130e


</details>

<details>
<summary>Mobile Web - Safari</summary>







https://github.com/Expensify/App/assets/46707890/5845b151-b88a-4bf9-9cee-e1607e77bfdf


</details>

<details>
<summary>Desktop</summary>







https://github.com/Expensify/App/assets/46707890/7da0058f-9740-4b2d-a928-121715c7b468


</details>

<details>
<summary>iOS</summary>






https://github.com/Expensify/App/assets/46707890/d1d691e0-8388-4f2a-9fa7-ef6749a78309


</details>

<details>
<summary>Android</summary>





https://github.com/Expensify/App/assets/46707890/0bb09945-fce9-4c20-90fe-4c6b89473bf5


</details>

**Comment by esh-g** [comment]
@abdulrahuman5196 Made the change for offline tests 💫

**Review by abdulrahuman5196 — APPROVED** [review]
Changes looks good and works well. Reviewers checklist is also complete.

All yours @aldo-expensify 

🎀 👀 🎀
C+ Reviewed

**Review by aldo-expensify — APPROVED** [review]
Tested well for me, thank you!

File: src/pages/workspace/WorkspacesListPage.js [hunk 1]
```diff
@@ -28,6 +28,7 @@ import * as ReimbursementAccountProps from '../ReimbursementAccount/reimbursemen
 import * as ReportUtils from '../../libs/ReportUtils';
 import * as CurrencyUtils from '../../libs/CurrencyUtils';
 import withPolicyAndFullscreenLoading from './withPolicyAndFullscreenLoading';
+import withWindowDimensions from '../../components/withWindowDimensions';
```
Comment (roryabraham) [inline src/pages/workspace/WorkspacesListPage.js:31]: Minor feedback for next time, we have hooks to replace many of our HOCs, and we should strive to use them going forward in functional components

- `withWindowDimensions` => `useWindowDimensions`
- `withLocalize` => `useLocalize`
- `withNetwork` => `useNetwork`

etc...

whenever refactoring a class component or adding new HOCs, please check `src/hooks` first. Thanks! 😄

**Pull request merged**